src/AzSK.ADO/Framework/Core/SVT/SVTResourceResolver.ps1

```diff
@@ -603,39 +597,55 @@ class SVTResourceResolver: AzSKRoot {
                             $this.PublishCustomMessage("Getting variable group configurations...");
                         }
 
-                        # Here we are fetching all the var grps in the project and then filtering out. But in build & release we fetch them individually unless '*' is used for fetching all of them.
-                        $variableGroupURL = ("https://dev.azure.com/{0}/{1}/_apis/distributedtask/variablegroups?api-version=6.1-preview.2") -f $($this.organizationName), $projectId;
-                        $variableGroupObj = [WebRequestHelper]::InvokeGetWebRequest($variableGroupURL)
-
-                        if (([Helpers]::CheckMember($variableGroupObj, "count") -and $variableGroupObj[0].count -gt 0) -or (($variableGroupObj | Measure-Object).Count -gt 0 -and [Helpers]::CheckMember($variableGroupObj[0], "name"))) {
+                        if ($this.VariableGroups -eq "*") {
+                            $variableGroupURL = ("https://dev.azure.com/{0}/{1}/_apis/distributedtask/variablegroups?api-version=6.1-preview.2" +$topNQueryString) -f $($this.organizationName), $projectId;
+                            $variableGroupObj = [WebRequestHelper]::InvokeGetWebRequest($variableGroupURL)
 
-                            $varGroups = $null;
-                            $projectData["variableGroups"] = ($variableGroupObj | Measure-Object).Count
-                            if ($this.VariableGroups -eq "*") {
-                                $varGroups = $variableGroupObj
-                            }
-                            else {
-                                #If service id based scan then filter with variablegroup ids
-                                if ($this.isServiceIdBasedScan -eq $true) {
-                                    $varGroups = $variableGroupObj | Where-Object { $this.VariableGroupIds -eq $_.Id }
+                            if (([Helpers]::CheckMember($variableGroupObj, "count") -and $variableGroupObj[0].count -gt 0) -or (($variableGroupObj | Measure-Object).Count -gt 0 -and [Helpers]::CheckMember($variableGroupObj[0], "name"))) 
```

Comment:
do we require this check : -or (($variableGroupObj | Measure-Object).Count -gt 0)

Reply:
Yes, otherwise it will break in some scenario

![image](https://user-images.githubusercontent.com/63220979/124604826-f60e8f80-de88-11eb-9ff4-c3df2f6fee6e.png)